pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,421 @@
+				o.ciDataClient,
+				o.ciGCSClient,
+				o.gcsClient,
+				"origin-ci-test",
+				(*o.jobGCSPrefixes)[i].gcsPrefix,
+			)
+		}
+
+		fmt.Printf("  launching findJobRunsWithRetry for %q\n", job.JobName)
+
+		waitGroup.Add(1)
+
+		go func() {
+			defer waitGroup.Done()
+			jobRuns, err := o.findJobRunsWithRetry(ctx, job.JobName, jobRunLocator)
+			if err == nil {
+				resultCh <- jobRuns
+			}
+		}()
+	}
+	waitGroup.Wait()
+	close(resultCh)
+
+	// drain the result channel first
+	for jobRuns := range resultCh {
+		jobRunsToReturn = append(jobRunsToReturn, jobRuns...)
+	}
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+		break
+	}
+	return jobRunsToReturn, nil
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) runTestCaseCheckers(ctx context.Context,
+	finishedJobRuns []jobrunaggregatorapi.JobRunInfo, unfinishedJobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestSuite {
+	suiteName := "payload-cross-jobs"
+	testSuite := &junit.TestSuite{
+		Name:      suiteName,
+		TestCases: []*junit.TestCase{},
+	}
+
+	allJobRuns := append(finishedJobRuns, unfinishedJobRuns...)
+	for _, checker := range o.testCaseCheckers {
+		testCase := checker.CheckTestCase(ctx, allJobRuns)
+		testSuite.TestCases = append(testSuite.TestCases, testCase)
+		testSuite.NumTests++
+		if testCase.FailureOutput != nil {
+			testSuite.NumFailed++
+		}
+	}
+	return testSuite
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) Run(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, o.timeout)
+	defer cancel()
+
+	matchID := o.payloadTag
+	if len(matchID) == 0 {
+		matchID = o.payloadInvocationID
+	}
+
+	outputDir := filepath.Join(o.workingDir, matchID)
+	if err := os.MkdirAll(outputDir, 0755); err != nil {
+		return fmt.Errorf("error creating output directory %q: %w", outputDir, err)
+	}
+
+	// if it hasn't been more than hour since the jobRuns started, the list isn't complete.
+	readyAt := o.jobRunStartEstimate.Add(1 * time.Hour)
+
+	// Our guess of the max time for jobs
+	durationToWait := o.timeout - 20*time.Minute
+	if durationToWait > (4*time.Hour + 15*time.Minute) {
+		durationToWait = 4*time.Hour + 15*time.Minute
+	}
+	timeToStopWaiting := o.jobRunStartEstimate.Add(durationToWait)
+
+	fmt.Printf("Analyzing test status for job runs for %q.  now=%v, ReadyAt=%v, timeToStopWaiting=%v.\n", matchID, o.clock.Now(), readyAt, timeToStopWaiting)
+
+	err := jobrunaggregatorlib.WaitUntilTime(ctx, readyAt)
+	if err != nil {
+		return err
+	}
+	finishedJobRuns, unfinishedJobRuns, _, _, err := jobrunaggregatorlib.WaitAndGetAllFinishedJobRuns(ctx, timeToStopWaiting, o, outputDir)
+	if err != nil {
+		return err
+	}
+
+	testSuite := o.runTestCaseCheckers(ctx, finishedJobRuns, unfinishedJobRuns)
+
+	// Done with all tests
+	junitXML, err := xml.Marshal(testSuite)
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(filepath.Join(outputDir, "junit-test-case-analysis.xml"), junitXML, 0644); err != nil {
+		return err
+	}
+	return nil
```

Comment:
if you have failures, don't you have to exit non-zero so that the job run will show up failed?  analyzer.go does this as I recall.